src/main/java/com/exasol/adapter/dialects/SqlGenerationVisitor.java

```diff
@@ -195,6 +198,40 @@ public String visit(final SqlJoin join) throws AdapterException {
                 + join.getRight().accept(this) + " ON " + join.getCondition().accept(this);
     }
 
+    @Override
+    public String visit(final SqlFunctionAggregateListagg sqlFunctionAggregateListagg) throws AdapterException {
+        final StringBuilder builder = new StringBuilder();
+        builder.append("LISTAGG(");
+        if (sqlFunctionAggregateListagg.isDistinct()) {
+            builder.append("DISTINCT ");
+        }
+        builder.append(sqlFunctionAggregateListagg.getArguments().get(0).accept(this));
+        if (sqlFunctionAggregateListagg.hasSeparator()) {
+            builder.append(", '");
+            builder.append(sqlFunctionAggregateListagg.getSeparator());
+            builder.append("'");
```

Comment:
I think this will fail, if the separator contains a `'` character. This seems to be a general problem occurring in a few places. There should be a function which generates a quoted string including the `'` characters around it:
```suggestion
            builder.append(", ");
            builder.append(quoteString(sqlFunctionAggregateListagg.getSeparator()));
```